### PR TITLE
Add fd.1 man page, close #77

### LINKED ISSUE #77: Add man (1) entry
Adding a man entry would be a nice improvement for when `fd` is packaged ([And possibly some day for when installed via `cargo install`?](https://github.com/rust-lang/cargo/issues/2729)).

### PR DESCRIPTION
**Pull request by pickfire**
https://github.com/sharkdp/fd/issues/77

### CODE REVIEW

File: doc/fd.1 [hunk 1]
```diff
@@ -0,0 +1,106 @@
+.TH FD 1 fd-4.0.0
+.SH NAME
+fd \- find files
```
Comment (sharkdp) [inline doc/fd.1]: Maybe this could be a little bit more detailed: "fd - find entries in the filesystem"?

File: doc/fd.1 [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+.BR find (1)
```
Comment (sharkdp) [inline doc/fd.1]: What is this for? is it a link to `find`? Do we need this here?

File: doc/fd.1 [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+.SH SYNOPSIS
+.B spt
```
Comment (sharkdp) [inline doc/fd.1]: This should be `fd`, probably :smile:

File: doc/fd.1 [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+.SH DESCRIPTION
+.B fd
+is a simple, fast and user-friendly alternative to
+.BR find (1).
+.SH OPTIONS
+.TP
+.B \-H, \-\-hidden
+Search hidden files and directories.
+.TP
+.B \-I, \-\-no\-gitignore
```
Comment (sharkdp) [inline doc/fd.1]: This should be `--no-ignore`.

File: doc/fd.1 [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+Do not respect any
+.BR gitignore (5)
+files.
+.TP
+.B \-s, \-\-case\-insensitive
```
Comment (sharkdp) [inline doc/fd.1]: This should be `--case-sensitive`

File: doc/fd.1 [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+.I pattern
+against full path.
+.TP
+.B \-0, \-\-print0
+Print results followed by a null character (instead of newlines).
+.TP
+.B \-h, \-\-help
+Prints help information.
+.TP
+.B \-v, \-\-version
```
Comment (sharkdp) [inline doc/fd.1]: This should be `-V`, not `-v`.

File: doc/fd.1 [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+Show colors only when standard output is connected to a terminal. (default)
+.IP never
+Do not use colorized output.
+.IP always
+Always use colorized output.
+.RE
+.TP
+.BI "\-j, \-\-threads " num
+Specifies number of threads to use for searching. (default: number of available
+CPU cores)
```
Comment (sharkdp) [inline doc/fd.1]: Can we also add the new option `-x`, `--exec`?

File: doc/fd.1 [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+.SH ENVIRONMENT
+.TP
+LS_COLORS
+.BR "fd"'s
+color codes for search results.
```
Comment (sharkdp) [inline doc/fd.1]: `LS_COLORS` is also used by `ls` and typically set by `dircolors`. Maybe we should say something like: "determines how to colorize the search results, see DIRCOLORS(1)".

File: doc/fd.1 [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+.SH EXAMPLES
+.TP
+.RI "Find the specific log files in " /var/log .
+$ fd nginx.log /var/log
```
Comment (sharkdp) [inline doc/fd.1]: Maybe just use `fd nginx /var/log` here. The `.` can actually mean "any character" in a regex, so this does not perform a literal search for "nginx.log".

File: doc/fd.1 (hunk 2 above)
Comment (pickfire) [reply doc/fd.1]: Yes, this is a link to find since it's on the current description.

File: doc/fd.1 (hunk 3 above)
Comment (pickfire) [reply doc/fd.1]: Ops, sorry. :P

File: doc/fd.1 (hunk 1 above)
Comment (pickfire) [reply doc/fd.1]: I took that from the posix man page and sbase man page. Both uses that.

File: doc/fd.1 (hunk 8 above)
Comment (pickfire) [reply doc/fd.1]: Nice suggestion, I don't have better idea to write this environment part.

**Comment by sharkdp** [comment]
Fantastic, thank you very much!

**Pull request merged**